sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2211,6 +2211,7 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         DecimalType(precision.getText.toInt, 0)
       case ("decimal" | "dec" | "numeric", precision :: scale :: Nil) =>
         DecimalType(precision.getText.toInt, scale.getText.toInt)
+      case ("void", Nil) => NullType
```

Comment:
If change here, spark will also support `void` type for table. Is it needed ?

Reply:
Just re-submit this PR since I don't know how to reopen. It's based on the latest discussions in #17953. Don't worry, we can close this if the community thinks the changes worthless to do.